code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -45,11 +44,13 @@
 			src << "<span class='boldannounce'>Your essence has dropped below critical levels. You barely manage to save yourself - [strikes ? "you can't keep this up!" : "next time, it's death."]</span>"
 		else if(strikes <= 0)
 			death()
-	maxHealth = essence * 2
+	maxHealth = essence * 3
 	if(!revealed)
 		health = maxHealth //Heals to full when not revealed
-	if(essence_regen && !inhibited && essence < essence_regen_cap) //While inhibited, essence will not regenerate
-		essence++
+	if(essence_regenerating && !inhibited && essence < essence_regen_cap) //While inhibited, essence will not regenerate
+		essence += essence_regen_amount
+		if(essence > essence_regen_cap)
+			essence = essence_regen_cap
```

Comment:
This won't actually fuck up and reduce your 400 essence to 25, because this bit only gets called if your essence is below 25.